dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractWorkflowResource.java

```diff
@@ -591,6 +591,9 @@ private Workflow addDockstoreYmlVersionToWorkflow(String repository, String gitR
             WorkflowVersion addedVersion = workflowVersionDAO.getWorkflowVersionByWorkflowIdAndVersionName(workflow.getId(), remoteWorkflowVersion.getName());
             if (addedVersion != null) {
                 gitHubSourceCodeRepo.updateVersionMetadata(addedVersion.getWorkflowPath(), addedVersion, workflow.getDescriptorType(), repository);
+                if (workflow.getLastModified() == null || workflow.getLastModifiedDate().before(addedVersion.getLastModified())) {
```

Comment:
Can `addedVersion.getLastModified()` return null? There are some null lastModified values in the DB, but maybe it never happens with new versions going forward. If it were null, I think `before` would NPE.